### PR TITLE
Make kube-state-metrics optional for installation

### PR DESCRIPTION
**Pull request by SergK**
In the most cases kube-state-metrics is already installed on the
cluster, so let's just re-use it, instead of installing another
one copy. By default, kube-state-metrics are installed as dependency,
so this PR doesn't change the default behaviour

Signed-off-by: Sergiy Kulanov <sergey@kulanov.org.ua>

- [*] Chart version *not* bumped (the versions are all bumped and released at the same time)
- [*] README.md updated with any new values or changes
- [*] Updated template tests in `${CHART}/tests/*.py` 
- [ ] Updated integration tests in `${CHART}/examples/*/test/goss.yaml`

### CODE REVIEW

**Comment by jmlrt** [comment]
jenkins test this please

File: metricbeat/templates/deployment.yaml [hunk 1]
```diff
@@ -92,7 +92,11 @@ spec:
             fieldRef:
               fieldPath: metadata.namespace
         - name: KUBE_STATE_METRICS_HOSTS
+          {{- if .Values.kube_state_metrics.enabled }}
           value: "$({{ .Release.Name | replace "-" "_" | upper }}_KUBE_STATE_METRICS_SERVICE_HOST):$({{ .Release.Name | replace "-" "_" | upper }}_KUBE_STATE_METRICS_SERVICE_PORT_HTTP)"
```
Comment (pbecotte) [inline metricbeat/templates/deployment.yaml]: Would it make more sense here to just use the kube_state_metrics_host value even if we are installing the sub-chart? In that case, what is the value that gets passed to the subchart as the service name?

Comment (SergK) [reply metricbeat/templates/deployment.yaml:90]: Hi @pbecotte . The general idea is to preserve upstream behavior as much as possible,
so when user disables `kube_state_metrics` installation then he/she needs to provide
existing endpoint for `kube_state_metrics`

**Comment by erihanse** [comment]
Is there something needed to get progress on this feature?

**Comment by jmlrt** [comment]
Hi @erihanse 

> Is there something needed to get progress on this feature?

The first thing is to have time for a proper review, tests and impacts analysis. This one is pretty old and important, it is in the top of the backlog and we should be able to review it soon.

@SergK, I'm sorry for the delay, Metricbeat had a lot of changes since you created this PR. Can you merge master on your branch and fix the conflicts?

**Comment by SergK** [comment]
@jmlrt done!

**Comment by jmlrt** [comment]
jenkins test this please

**Comment by jmlrt** [comment]
Some changes are required to make [Black](https://black.readthedocs.io/en/stable/) formatter happy in [elastic+helm-charts+pull-request+lint-python/361](https://devops-ci.elastic.co/job/elastic+helm-charts+pull-request+lint-python/361/label=docker&&virtual&&ubuntu-18.04/console).

Can you run `black tests/metricbeat_test.py` and update your PR?

**Comment by SergK** [comment]
Sure!

**Comment by jmlrt** [comment]
jenkins test this please

File: metricbeat/values.yaml [hunk 2]
```diff
@@ -250,6 +250,12 @@ updateStrategy: RollingUpdate
 nameOverride: ""
 fullnameOverride: ""
 
+# is used only when kube_state_metrics.enabled: false
+kube_state_metrics_host: ""
```
Comment (jmlrt) [inline metricbeat/values.yaml:254]: What about having `kube_state_metrics.host` instead of `kube_state_metrics_host`?

**Comment by jmlrt** [comment]
jenkins test this please

**Comment by jmlrt** [comment]
backported to `6.8`, `7.8`and `7.x` branches

**Pull request merged**